pkg/kapp/preflight/registry.go

```diff
@@ -101,9 +102,56 @@ func (c *Registry) AddCheck(name string, check Check) {
 	if c.known == nil {
 		c.known = make(map[string]Check)
 	}
+	if c.enabledFlag == nil {
+		c.enabledFlag = make(map[string]bool)
+	}
 	c.known[name] = check
 }
 
+// Validate the configuration provided; the rules are:
+// 1. Unknown validator = error
+// 2. Duplicate validator = error
+func (c *Registry) validateConfig(conf config.Conf) error {
+	haveConfig := map[string]bool{}
+	for _, rule := range conf.PreflightRules() {
+		if _, ok := c.known[rule.Name]; !ok {
+			return fmt.Errorf("unknown preflight check in configuration: %q", rule.Name)
+		}
+		if _, ok := haveConfig[rule.Name]; ok {
+			return fmt.Errorf("duplicate preflight check in configuration: %q", rule.Name)
+		}
+		haveConfig[rule.Name] = true
+	}
+	return nil
+}
+
+func (c *Registry) SetConfig(conf config.Conf) error {
```

Comment:
_Maybe_. I understand you want to get rid of the YAML in the unit tests, but I would counter that if there are errors parsing the YAML, it would be due to a change _that we might have to take into consideration_. So, would it not be better to have an error in this case: the author of the change that "breaks" the YAML parsing would see this immediately.

It's not necessarily fragile, and it allows us to fully test the configuration from the YAML source. 

I would also state that the configuration YAML is likely to be easier to read than the golang-formatted structure.

Reply:
>So, would it not be better to have an error in this case: the author of the change that "breaks" the YAML parsing would see this immediately.

IMO it is better for tests explicitly associated with the `config` package to rather than packages that are provided values from the `config.Conf` Go type.

I don't mind having the YAML in the unit tests if it is absolutely necessary, but IMO the parsing of YAML has absolutely nothing to do with the logic for the SetConfig function - in this case, it is assumed that it _already_ has been parsed. IMO we should be testing based on the expectations we have and not adding points of failure beyond the logic within the "unit" we are testing.

At the end of the day this isn't a hill I'm willing to die on. This conversation in particular has held this PR up long enough so what you have is fine. I'm going to do another sweep to make sure I didn't miss anything else